src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php

```diff
@@ -139,13 +144,69 @@ public function load(array $configs, ContainerBuilder $container)
         ));
     }
 
+    private function getBundleHierarchy(ContainerBuilder $container)
+    {
+        $bundleHierarchy = array();
+
+        foreach ($container->getParameter('kernel.bundles_metadata') as $name => $bundle) {
+            if (!array_key_exists($name, $bundleHierarchy)) {
+                $bundleHierarchy[$name] = array(
+                    'paths' => array(),
+                    'parents' => array(),
+                    'children' => array(),
+                );
+            }
+
+            if (is_dir($dir = $container->getParameter('kernel.root_dir').'/Resources/'.$name.'/views')) {
```

Comment:
Note to maintainers: when merging to newer branches, we need to take care of the FileExistenceResource objects here (git conflicts should remind us about this btw)